### DISCUSSION TITLE
[linux-x86] Is it possible to cross build the coreclr or mono runtime?

### DISCUSSION THREAD

**Discussion by ikkentim** [dotnet/runtime#66963] [Q&A]
I'm trying to get a runtime to work on linux-x86. I've first tried to get the coreclr running, but I ran in various SIGSGV... and I don't really know how to debug it.

I thought maybe it would be possible to build the mono runtime for x86, but when i try to cross build it the output binaries seem to be x64.

Could someone tell me if I'm doing something wrong?

```
sudo ROOTFS_DIR=/home/tim/runtime/.tools/rootfs/x86 ./build.sh --subset mono+libs --arch x86 -c release --cross

readelf -a -W  artifacts/bin/microsoft.netcore.app.runtime.linux-x86/Release/runtimes/linux-x86/native/libcoreclr.so | more
```

```
ELF Header:
  Magic:   7f 45 4c 46 02 01 01 00 00 00 00 00 00 00 00 00
  Class:                             ELF64
  Data:                              2's complement, little endian
  Version:                           1 (current)
  OS/ABI:                            UNIX - System V
  ABI Version:                       0
  Type:                              DYN (Shared object file)
  Machine:                           Advanced Micro Devices X86-64
```

**Comment by am11** [comment]
> I've first tried to get the coreclr running, but I ran in various SIGSGV.

Please open issues with details about those SIGSEGV in coreclr linux-x86. It is a community supported platform, known to be working at some point. It is likely an unintended regression by a (recent) code change.

> I thought maybe it would be possible to build the mono runtime for x86

Unlike mono/mono repo, the build configuration for linux-x86 is currently not set up for mono in runtime repo yet.

I think it should be possible to add those configurations without any code changes. In`src/mono/mono.proj`, search for `DCMAKE_TOOLCHAIN_FILE` and adapt those blocks for `Condition="'$(TargetsLinux)' == 'true' '$(TargetArchitecture)' == 'x86'"` and see if there is any other change needed in `src/mono/CMakeLists.txt`.
